tool/bindings_generator/translator.dart

```diff
@@ -453,15 +463,32 @@ class Translator {
     }
   }
 
-  code.TypeReference _typeReference(String symbol,
-      {bool isNullable = false, bool isReturn = false}) {
+  // Given a raw type, convert it to the Dart type that will be emitted by the
+  // translator.
+  code.TypeReference _typeReference(_RawType type) {
+    var dartType = type.type;
+    var nullable = type.nullable;
+
+    // Convert JS types to primitives.
+    if (dartType == 'JSBoolean') {
```

Comment:
style nit: this may be a nice use of the new pattern syntax :)

```dart
dartType = switch(dartType) {
  'JSBoolean' => 'bool',
  'JSString' => 'String',
  'JSInteger' => 'int',
  'JSDouble' => 'num',
  'JSNumber' => 'num',
  'JSUndefined' => 'void',
  _ => dartType,
};
if (dartType == 'void') nullable = false;
```

Reply:
Done, thanks!